### ISSUE TITLE
Device is always discoverable via Bluetooth

### ISSUE THREAD

**Issue by JamiKettunen** [HelloVolla/ubuntu-touch-beta-tests#91]
**Details about installed OS version**
Applies to all Ubuntu Touch builds for the Volla Phone.

**Describe the bug**
While Bluetooth can be disabled in the UI thanks to the rfkill kernel stack, it appears that the BT driver itself isn't aware of rfkill **at all**, and the UI animation of it turning off has been purely placebo.

I can confirm this by turning off BT and scanning for devices from another nearby device, which sshows `ubuntu-phablet` as always visible. While pairing can be attempted in this placebo off state nothing seems to happen on either of the devices.

**Steps to reproduce the bug**
1. Disable Bluetooth on Volla Phone
2. Scan for BT devices from another device
3. See `ubuntu-phablet` (Volla Phone) always visible in the results no matter how long has been waited since BT was supposedly disabled on the Volla Phone.

**Expected behavior**
Bluetooth is disabled (powered off) on the Volla Phone and it can no longer be discovered on other devices that initiate a scan afterwards.

**Additional context**
~~Likely [the kernel driver](https://github.com/HelloVolla/android_kernel_volla_mt6763/tree/halium-9.0/drivers/misc/mediatek/connectivity/bt/mt66xx) expects Android to do something special here as [rfkill was fully disabled by default in kernel configuration](https://github.com/HelloVolla/android_kernel_volla_mt6763/commit/8d82bec2ca7e99019fed79e4977efbcb1acda526).~~

As the same issue has been confirmed as present on a OnePlus 5T (Qualcomm MSM8998 SoC) under Sailfish OS, it's pretty safe to say this is more likely a [bluebinder](https://github.com/mer-hybris/bluebinder) issue insstead.

**Comment by gbdomubpkm** [comment]
What is the state of this issue in dev ?
Does the bug still occur in dev  (or RC or OTA-17) ?

I tasted using two Volla phones, one with volla os, the other with UT in dev 382.

After activating bluetooth in Volla UT and after toggling off bluetooth, i don't scan the Volla UT (ubuntu-phablet) on the Volla VOLLA OS which has bluetooth on.

**Comment by gbdomubpkm** [comment]
What is the status of this issue with OTA-22 ?

**Comment by IanHasman** [comment]
Is this a duplicate or at least related?

https://github.com/HelloVolla/ubuntu-touch-beta-tests/issues/119

**Comment by JamiKettunen** [comment]
This appears to still be an issue on dev channel `16.04 (2022-07-08)`.

If you enable bluetooth **and** go to the Bluetooth settings page the device becomes and stays discoverable **even if you disable Bluetooth** until you either:
- Wait 3 minutes **with Bluetooth enabled** until the "seemingly invisible" discoverable time runs out (this should really be configurable or at least visible in the Bluetooth settings page)
- Reboot the device
- `sudo service bluebinder restart` (and wait ~30 seconds or so)

I'm going to try out https://github.com/mer-hybris/bluebinder/pull/29 and see if that fixes this problem

**Comment by JamiKettunen** [comment]
The above pull request fixes this issue! Need to wait for it to get merged, a new upstream tag to be created and UBports importing the new changes :tada:

**Comment by JamiKettunen** [comment]
The fix is now submitted as https://github.com/ubports/bluebinder/pull/8, @gbdomubpkm @IanHasman please try the following :)
```
ubports-qa install bluebinder 8
```

**Comment by gbdomubpkm** [comment]
Hello @JamiKettunen 
I installed and everything seems ok, even after a restart of the Volla.
Can we change the name 'ubuntu-phablet' ? or is it akways this name for all Volla ?
![screenshot20220714_165237772](https://user-images.githubusercontent.com/50362604/179012100-a4ec6390-4a51-4a85-a280-1b3a135de1eb.png)

**Comment by JamiKettunen** [comment]
>Can we change the name 'ubuntu-phablet' ? or is it akways this name for all Volla ?

@gbdomubpkm for now you can use `bluetoothctl` and do e.g. `system-alias Volla Phone` to change the visible name seen by other devices, but the System Settings will keep insisting it's `ubuntu-phablet` even if it's not really discoverable as that anymore; ideally this should be easily editable from the Bluetooth system-settings page

**Comment by JamiKettunen** [comment]
With https://gitlab.com/ubports/development/core/lomiri-system-settings/-/merge_requests/340 I can see the `system-alias` set in `bluetoothctl` in System Settings as well; unfortunately testing this change isn't as easy as `ubports-qa install lomiri-system-settings 340` simply doesn't appear to work for whatever reason saying `CRITICAL:ubports-qa:Jenkins build not found` despite [the artifacts clearly existing](https://ci.ubports.com/blue/organizations/jenkins/UBportsCore%2Flomiri-system-settings/detail/MR-340/1/artifacts/)

**Comment by gbdomubpkm** [comment]
I have done more tests and i have bad news :  the device is always discoverable even if bluetooth is shown off in system settings of the UT Volla, which is detected by my other Volla with Volla OS !!

For bluetoothctl system-alias, it works very well : my vollaphone with volla os detected the ut volla with the new name and no problem to connect the first with the second.

**Comment by JamiKettunen** [comment]
> I have done more tests and i have bad news : the device is always discoverable even if bluetooth is shown off in system settings of the UT Volla, which is detected by my other Volla with Volla OS !!
> 
> Please can you verify that your patch installs correctly and completely. I have a doubt about this. Maybe I'm wrong..

@gbdomubpkm I've rather extensively tested the commit which is included in that `bluebinder` PR to fix the issue, if you can still "discover" the device especially on Android go back in the Bluetooth settings page and click `Pair new device` again, it appears to "cache" the found devices even if they're not discoverable anymore until you visit the same settings page again.

**Comment by JamiKettunen** [comment]
@gbdomubpkm it really doesn't sound like you have the bluebinder PR package installed as what you're seeing mimmicks the exact behavior I saw with the old version, please confirm the following:
```sh
apt show bluebinder | grep "^Version"
# this should return "1.0.1+0~20220714140105.1~1.gbp21a187"
# otherwise please do the following:
ubports-qa list
# confirm the list shows "PR_bluebinder_8"
sudo ubports-qa remove PR_bluebinder_8
# reinstall the fixed package
sudo ubports-qa install bluebinder 8
# reboot the device
```

I also finally found a way to install the alias MR I mentioned being more difficult to test: `ubports-qa install PR_lomiri-system-settings_340`

**Comment by gbdomubpkm** [comment]
![screenshot20220715_141600011](https://user-images.githubusercontent.com/50362604/179221569-a1859438-8c9f-41f7-b049-8c2a3356ad24.png)

**Comment by JamiKettunen** [comment]
@gbdomubpkm if you were scanning for this phone with new `bluebinder` from another Ubuntu Touch device it appear the scan result caching on UT is even more aggressive, I can still see the phone still long after powering it off entirely and the list doesn't seem to get refreshed by even re-opening the settings app either or even turning off Bluetooth on the searching device.

I was properly able to test the discovery on Android (by going back and `Pair new device` again, the phone was no longer seen in a new fresh search) and from a laptop running Fedora 36 with KDE.

**Comment by JamiKettunen** [comment]
You can try, but it looks installed to me. Try scanning for the device on a non-Ubuntu Touch device or even computer as I already stated UT has really aggressive caching of the results.

**Comment by gbdomubpkm** [comment]
So I'm not doing anything at the moment. As I told you, the Volla ut was first scanned by a Volla volla OS... You would have to be able to confirm my statements with a Volla volla OS.

**Comment by gbdomubpkm** [comment]
I reinstalled and i got
![screenshot20220715_175406019](https://user-images.githubusercontent.com/50362604/179261701-40af3f78-ddd5-4688-8e41-cd54ad79d54c.png)
Is it good/normal ?

**Comment by gbdomubpkm** [comment]
On my BQ E5 HD, i had to use 
`[bluetooth]# remove 00:00:00:00:00:00` in terminal  to clear old devices (using their Mac address).

**Comment by gbdomubpkm** [comment]
> I also finally found a way to install the alias MR I mentioned being more difficult to test: ubports-qa install PR_lomiri-system-settings_340

Is it an easy way ? I tried with sudo and even if the package seemed installed, it didn't seem to do anything in system settings. Am i wrong ?

**Comment by JamiKettunen** [comment]
This was tracked on https://ubports.atlassian.net/browse/YUMI-16 and the fix (new `bluebinder` version) should now be available on both latest 16.04 development update channel and 20.04 on all models :)
